vyper/venom/__init__.py

```diff
@@ -56,10 +55,30 @@ def _run_passes(ctx: IRFunction, optimize: OptimizationLevel) -> None:
     for entry in internals:
         SimplifyCFGPass().run_pass(ctx, entry)
 
+    dfg = DFG.build_dfg(ctx)
+    Mem2Var().run_pass(ctx, ctx.basic_blocks[0], dfg)
+    for entry in internals:
+        Mem2Var().run_pass(ctx, entry, dfg)
+
     make_ssa_pass = MakeSSA()
```

Comment:
can MakeSSA() reuse the dominator tree from Mem2Var()?

Reply:
We could but the plan is to let the pass manager handle those kinds of "optimization" automatically. Therefore no effort currently being applied to this pass sequencing, that is constantly evolving as we shape the passes to their final form.